packages/fxa-auth-server/scripts/populate-firestore-customers/firestore-populator.ts

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { accountExists } from 'fxa-shared/db/models/auth';
+import { Stripe } from 'stripe';
+
+import { StripeHelper } from '../../lib/payments/stripe';
+import { StripeFirestore } from '../../lib/payments/stripe-firestore';
+import { RateLimitObserver } from './rate-limit-observer';
+
+export class FirestorePopulator {
+  rateLimitObserver: RateLimitObserver;
+  stripe: Stripe;
+
+  constructor(
+    private log: any,
+    private stripeHelper: StripeHelper,
+    rateLimit: number
+  ) {
+    this.stripe = this.stripeHelper.stripe as Stripe;
+    // Hook up the rate limit observer
+    this.rateLimitObserver = new RateLimitObserver(rateLimit);
+    this.stripe.on('request', (request) => this.rateLimitObserver.increment());
+  }
+
+  public async populate() {
+    const stripeFirestore = (this.stripeHelper as any)
+      .stripeFirestore as StripeFirestore;
+
+    for await (const stripeUser of this.stripe.customers.list({ limit: 100 })) {
```

Comment:
Maybe add a try-catch?  Some of the code below could throw an error.

Reply:
That's true, but in that case.... we'd need to re-run the entire thing to go and hit the ones that didn't work. The firestore fetch before attempting to update means that we'll breeze through no-ops with only a single Stripe call per 100 batch, so it should catch up relatively quick.